Automattic/jetpack
# Gutenberg: load blocks from CDN

A suggestion for easy testing of Gutenberg Blocks in Jetpack.

This PR doesn't explore how to _build_ these blocks, but how to _load_ them in Jetpack in one file.

This should be flexible enough to allow very easy testing as well as easier local development by allowing to disable CDN.

Note that this contains only `editor` side assets.

#### Testing instructions:
1. Apply D16998-code to your sandbox to get `https://s0.wp.com/wp-content/mu-plugins/jetpack/_inc/blocks/jetpack-editor.js`
2. Confirm that by default nothing happens in Gutenberg editor
3. Confirm that you can enable these blocks via filter:
    ```php
    add_filter( 'jetpack_gutenberg', '__return_true', 10 );
    ```
    (You can place this e.g. in a file at `mu-plugins` -folder)
4. Confirm that by default files are loaded via CDN, but that you can disable this via filter:
    ```php
    add_filter( 'jetpack_gutenberg_cdn', '__return_false', 10 );
    ```
    Note that you would need to build files manually by running this in Calypso:
    ```
    npm run sdk:gutenberg -- \
	    --editor-script=client/gutenberg/extensions/presets/jetpack/editor.js \
	    --output-dir=~/jetpack/_inc/blocks
    ```
5. Confirm that by default CDN has a date +Jetpack version as a cachebuster, but that you can change it by filter:
    ```php
    add_filter( 'jetpack_gutenberg_cdn_cache_buster', function( $version ) { return time(); }, 10, 1 );
    ```

Blocks that you should have in Gutenberg editor:
- Tiled gallery
- Markdown
- Publicize

## TODO

- I had issues getting blocks working but it might be due to some changes in blocks and/or Gutenberg — I need to have a closer look (ping @tyxla)
- CSS file doesn't exist yet, it's waiting for https://github.com/Automattic/wp-calypso/pull/26551
